### DISCUSSION TITLE
Localization / translation of the modal content

### DISCUSSION THREAD

**Discussion by VladLegkowski** [mercadopago/sdk-js#193] [General]
Hi, using your sandbox I cannot launch a modal with a selected locale.

Code:

```
const mercadopago = new MercadoPago('KEY', {
  locale: 'pt-BR'
});
```

I get the button initialised in `pt`:
![image](https://github.com/mercadopago/sdk-js/assets/17318500/4c55386e-d2f5-4858-a3a1-221d2d9dd076)

But the actual modal is launched in `es`:
![image](https://github.com/mercadopago/sdk-js/assets/17318500/2df7fe84-57db-491b-807e-053b86ff8840)

Am i missing something in configuration?

Thanks.

**Comment by icaldana** [comment]
Hello @VladLegkowski,

For what I can see you are integrating Wallet Brick, right? Are you generating the preference with your PUBLIC_KEY?

Thanks!

**Reply by VladLegkowski** [reply]
Hello, thanks for the reply! yes I am, I am using your [example from here](https://github.com/mercadopago/checkout-payment-sample/blob/master/client/html-js/js/index.js). I am using public key.

**Reply by icaldana** [reply]
Hi @VladLegkowski, I far as I know the modal language is associated with the language of your `PUBLIC_KEY`. I am trying to confirm if there is a way to specify a different language during preferences' creation, but part of the team is on holidays so I will have the final response only about next week. I hope you understand.

Happy holidays!

**Reply by VladLegkowski** [reply]
Thanks a lot, I understand the timing is not great. But interesting that the locale only applies to the `Launch button`, but does not apply to the interface of the checkout. 

Thanks anyways and I look forward hearing from you soon.

**Reply by icaldana** [reply]
@VladLegkowski hello. I've got the confirmation from the responsible team and they confirmed the language and text are based on the start URL domain. Thus, for now there is no way to customize based on preference.

We added your request on our backlog and we appreciate your comprehension. Thank you!

**Reply by nttcarlosabia** [reply]
@icaldana hello, thank you for your response. We have a public key generated on https://www.mercadopago.com.ar/settings/account/applications/create-app. That public key gives us the AR-es language. Then do we need to generate a public key for each language and market?

**Reply by icaldana** [reply]
Hello @nttcarlosabia! As mentioned, it seems the language is based on the start URL domain, but if you found a way to specify the language during PK creation, you might want to validate it does not affect the modal texts.
